src/components/ContactPreview.tsx

```diff
@@ -0,0 +1,50 @@
+'use client'
+
+import { css } from '../../styled-system/css'
+import useAvatar from '@/components/useAvatar'
+import { useRouter } from 'next/navigation'
+import { createClientComponentClient } from '@supabase/auth-helpers-nextjs'
+import { Database } from '@/types/supabase'
+import { useContext } from 'react'
+import { UserContext } from '@/components/UserProvider'
+import { Profile } from '@/types/models'
+
+type Props = {
+  contact: Profile
+}
+
+export default function ContactPreview({ contact }: Props) {
+  const supabase = createClientComponentClient<Database>()
+  const router = useRouter()
+  const { user } = useContext(UserContext)
+
+  const { AvatarPreview } = useAvatar({
+    avatarUrl: contact.avatar_url,
+  })
+
+  const handleClick = async () => {
+    // TODO: check the chatId not to create multiple chats for the same users
+    const { data, error } = await supabase
+      .from('chats')
+      .insert({
+        users: [user?.id!, contact.id],
+      })
+      .select()
+      .single()
+    if (error) {
+      return
+    }
+    router.push(`/messages/${data.chat_id}`)
+  }
+
+  return (
+    // onClick on a div 🤔
+    <div className={css({ display: 'flex' })} onClick={handleClick}>
+      <AvatarPreview
+        className={css({ borderRadius: '50%', mr: 4 })}
+        size={60}
+      />
+      <div>{contact.username}</div>
```

Comment:
bug: When I loggin [this is what I see](https://github.com/alexodan/chat-app/assets/14869087/0cf3330b-aeb2-4f94-98db-009e3045b8fa). Some users don't have avatar or username. It would be nice to have fallback (placeholder and email) or just remove those.

Reply:
I made the username required when registering, also added a placeholder avatar image